sktime/forecasting/hf_transformers_forecaster.py

```diff
@@ -251,15 +270,46 @@ def _predict(self, fh, X=None):
         self.model.eval()
         from torch import from_numpy
 
-        hist = self._y.values.reshape((1, -1))
+        if isinstance(self._y.index, pd.MultiIndex):
+            lens = self._y.groupby(
+                level=list(range(len(self._y.index.levels) - 1))
+            ).apply(lambda x: len(x))
+            assert (lens == lens.iloc[0]).all(), "All series must has the same length"
+            hist = self._y.values.reshape((-1, lens.iloc[0]))
+        else:
+            hist = self._y.values.reshape((1, -1))
         if X is not None:
-            hist_x = self._X.values.reshape((1, -1, self._X.shape[-1]))
-            x_ = X.values.reshape((1, -1, self._X.shape[-1]))
-            if x_.shape[1] < self.model.config.prediction_length:
-                # TODO raise exception here?
-                x_ = np.resize(
-                    x_, (1, self.model.config.prediction_length, x_.shape[-1])
+            if isinstance(self._y.index, pd.MultiIndex):
+                hist_x = np.array(self._X.values, dtype=np.float32).reshape(
+                    (-1, lens.iloc[0], self._X.shape[-1])
+                )
+                lens = X.groupby(
+                    level=list(range(len(self._y.index.levels) - 1))
+                ).apply(lambda x: len(x))
+                assert (
+                    lens == lens.iloc[0]
+                ).all(), "All series must has the same length"
+                x_ = np.array(X.values, dtype=np.float32).reshape(
+                    (-1, lens.iloc[0], self._X.shape[-1])
                 )
+                if x_.shape[1] < self.model.config.prediction_length:
+                    # TODO raise exception here?
```

Comment:
@fkiraly is there an exception raised if there are not enough exogenous data passed for a specific forecasting horizon?